runtime/src/main/java/org/corfudb/runtime/view/stream/StreamAddressSpace.java

```diff
@@ -216,6 +263,8 @@ public StreamAddressSpace getAddressesInRange(StreamAddressRange range) {
 
     public void setTrimMark(long trimMark) {
         this.trimMark = trimMark;
+        // TODO(Maithem): This method should be removed. When the trimMark is set here, then trim is called
```

Comment:
But before removing this, changes need to be done on the trim method. The way it is right now, it would return when no addresses are in the bitmap without setting the trim mark (and this is needed as it is used when rebuilding the stream address space on the log unit)

Reply:
Tried to do it in this PR, but a bunch of tests failed. I will address that in a subsequent PR.

Reply:
Do those tests fail if you call trim here?